lib/framework/permissions_manager.dart

```diff
@@ -11,33 +14,63 @@ class PermissionsManager {
 
   Future<bool?> hasPermission(Permission type) async {
     bool? status;
-    if (type == Permission.notification) {
-      // firebase messaging does NOT work on Web's html-renderer
-      if (kIsWeb) {
-        status = await notificationUtils.hasPermission();
-      } else {
-        // use Firebase for status check all other platforms
-        var settings =
-            await FirebaseMessaging.instance.getNotificationSettings();
-        switch (settings.authorizationStatus) {
-          case AuthorizationStatus.authorized:
-          case AuthorizationStatus.provisional:
+
+    switch (type) {
+      case Permission.notification:
+        // firebase messaging does NOT work on Web's html-renderer
+        if (kIsWeb) {
+          status = await notificationUtils.hasPermission();
+        } else {
+          // use Firebase for status check all other platforms
+          var settings =
+              await FirebaseMessaging.instance.getNotificationSettings();
+          switch (settings.authorizationStatus) {
+            case AuthorizationStatus.authorized:
+            case AuthorizationStatus.provisional:
+              status = true;
+              break;
+            case AuthorizationStatus.denied:
+              status = false;
+              break;
+            case AuthorizationStatus.notDetermined:
+            default:
+              status = null;
+              break;
```

Comment:
We should move this location specific logic into the utilities or this class will become very large.